jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java

```diff
@@ -102,6 +105,38 @@
         appRoot, rawConfiguration.getJvmFlags(), mainClass);
   }
 
+  /**
+   * Checks whether or not current Java version is compatible with the base image.
+   *
+   * @param baseImageConfiguration the configured base image
+   * @param fromImageParameterName the name of the base image configuration parameter
+   * @throws InvalidImageReferenceException if the project is incompatible with the base image
```

Comment:
I don't think `InvalidImageReferenceException` is the right exception use here since that refers to an invalid image reference for parsing, not the actual contents of the image in relation to the proper runtime.